arangod/Aql/CollectionAccessingNode.cpp

```diff
@@ -67,7 +70,15 @@ CollectionAccessingNode::CollectionAccessingNode(
   VPackSlice restrictedTo = slice.get("restrictedTo");
 
   if (restrictedTo.isString()) {
-    _restrictedTo = restrictedTo.copyString();
+    auto maybeShardID = ShardID::shardIdFromString(restrictedTo.copyString());
+    if (ADB_UNLIKELY(maybeShardID.fail())) {
+      LOG_TOPIC("63e33", WARN, Logger::AQL)
+          << "Internal JSON representation of CollectionAccess for collection "
+          << colName << " contains invalid shard ID "
+          << restrictedTo.copyString() << " - ignoring";
```

Comment:
We input parameter of `ShardID::shardIdFromString()` should be an std::string_view, not an std::string.
Then this code part can be made more efficient as:
```suggestion
    auto maybeShardID = ShardID::shardIdFromString(restrictedTo.stringView());
    if (ADB_UNLIKELY(maybeShardID.fail())) {
      LOG_TOPIC("63e33", WARN, Logger::AQL)
          << "Internal JSON representation of CollectionAccess for collection "
          << colName << " contains invalid shard ID "
          << restrictedTo.stringView() << " - ignoring";
```

Reply:
Fixed by: 553ed9e25e7